GVFS/GVFS.Common/Cleanup/GitCleanupStep.cs

```diff
@@ -94,9 +94,10 @@ public void Execute()
 
         public void Stop()
         {
-            this.Stopping = true;
             lock (this.gitProcessLock)
             {
+                this.Stopping = true;

```

Comment:
@derrickstolee  note this small change to put this.Stopping in the lock I believe fixes a race condition between this function and CreateProcessAndRun.

- Stop: could have been called but not yet set "Stopping = true"
- CreateProcessAndRun: could have passed the this.Stopping check but not yet set the GitProcess
- Stop: could still have this.GitProcess == null and noop
- CreateProcessAndRun: could then go on to the set Process and run

Let me know if you think I missed something.  Threads can be tricky!